### ISSUE TITLE
Switch to next AMS spool when filament is empty

### ISSUE THREAD

**Issue by MrD-RC** [bambulab/BambuStudio#2075]
It would be a nice feature if you could switch to the next (or specified) slot, when the filament in the current spool runs out. This should be optional, and disabled by default.

For example, I'm printing a large project, which will use up the remaining filament on the spool in slot 1. It would be nice to be able to specify a slot to automatically use once the current spool empties. A simpler option would be to just use the next slot. But selecting a slot would be much more customisable. Maybe you could even present the option if Bambu Studio knows the print uses 700g of filament, for example. But the selected slot has less filament available on it.

**Comment by Hologos** [comment]
This feature is already in there, it's called auto switch filament.

**Comment by MadMax389** [comment]
> This feature is already in there, it's called auto switch filament.

Yes, but only if you remember to set up the AMS beforehand. It would be nice, as OP requested, to be prompted to switch to any slot with the same material when it runs out.

**Comment by Hologos** [comment]
@MadMax389 Sorry, somehow missed that. +1 it would be cool

**Comment by ChrisVonFrosta** [comment]
1 year later. Somthing happend? I still run in this error each role. But from remote i can't resume the print, even not with other color

**Comment by morsik** [comment]
It's 2026 and issue remains.

I've had tangled spool which I untangled and put (without roll with RFID part) into Slot 1. Because it was without RFID, I had to set filament manually.

On Slot 2 I put exact same spool this time with RFID. It was detected.

But auto-switch filament didn't worked because apparently manually setting "Bambu PLA Black" is not the same as "Bambu PLA Black" from RFID (WHY?!) and printer refuses to continue even it had exact same filament put into AMS and even set in settings... :/

There should be simple button: **"Force another filament, I'm aware of possible problems"**.

**Comment by ryclorak** [comment]
I ended up just cutting the RFID tags off, shoved one of the old tags into the new spool, and it's finally printing. 

At the very least, the RFID tag and manual settings should be recognized as the same, especially if/when the RFID reader malfunctions on one of the slots. However, it would still be better to have a force override option.

**Comment by mmmankey** [comment]
Bought an AMS because I assumed this was already feature. I've just recently learned that it isn't. Auto-refill is nice but it has to be the exact filament which is inconvenient when I have my AMS filled with different filaments.
